cmd/ztest/Makefile.am

```diff
@@ -2,7 +2,7 @@ include $(top_srcdir)/config/Rules.am
 
 # -Wnoformat-truncation to get rid of compiler warning for unchecked
 #  truncating snprintfs on gcc 7.1.1.
-AM_CFLAGS += $(DEBUG_STACKFLAGS) $(FRAME_LARGER_THAN) $(NO_FORMAT_TRUNCATION)
+AM_CFLAGS += $(DEBUG_STACKFLAGS) $(FRAME_LARGER_THAN) $(NO_FORMAT_TRUNCATION) -UNDEBUG
 AM_CPPFLAGS += -DDEBUG
```

Comment:
Can you move this down in to `AM_CPPFLAGS` with `-DDEBUG` since it is a preprocessor flag.